### PR TITLE
[DESK] Show monitor on "Settings" page when only one monitor detected

### PR DESCRIPTION
**Pull request by binarymaster**
Hide multiple monitor selection controls in this case. Otherwise, show them when multiple monitors are detected and hide the monitor preview bitmap.

This fixes the focus on the resolution slider when tested in Windows XP.

Also use the monitor bitmap globally, because it's used on multiple pages. I will also add it to the "Screensaver" page once this PR is merged.

**JIRA issues: [CORE-17939](https://jira.reactos.org/browse/CORE-17939) and [CORE-10606](https://jira.reactos.org/browse/CORE-10606)**

## Before

VirtualBox_Windows XP_01_10_2022_16_00_33

## After

VirtualBox_Windows XP_01_10_2022_16_08_48 VirtualBox_Windows XP_01_10_2022_16_08_55

### CODE REVIEW

File: dll/cpl/desk/desk.c [hunk 1]
```diff
@@ -197,6 +198,16 @@ DisplayApplet(HWND hwnd, UINT uMsg, LPARAM wParam, LPARAM lParam)
     g_GlobalData.pwszAction = pwszAction;
     g_GlobalData.desktop_color = GetSysColor(COLOR_DESKTOP);
 
+    /* Initialize the monitor preview bitmap, used on multiple pages */
+    g_GlobalData.hMonitorBitmap = (HBITMAP) LoadImage(hApplet, MAKEINTRESOURCE(IDC_MONITOR), IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
```
Comment (katahiromz) [inline dll/cpl/desk/desk.c:202]: `LoadBitmapW`?

File: dll/cpl/desk/settings.c [hunk 2]
```diff
@@ -421,6 +426,40 @@ SettingsOnInitDialog(IN HWND hwndDlg)
...
+    hOldObj = SelectObject(hDC, g_GlobalData.hMonitorBitmap);
+
+    /* FIXME: Draw resolution preview bitmap inside monitor. */
+    hBrush = CreateSolidBrush(g_GlobalData.desktop_color);
+    FillRect(hDC, &rcItem, hBrush);
+    DeleteObject(hBrush);
+
+    GdiTransparentBlt(draw->hDC,
+                      draw->rcItem.left, draw->rcItem.top,
+                      draw->rcItem.right-draw->rcItem.left+1,
```
Comment (katahiromz) [inline dll/cpl/desk/settings.c:452]:
```suggestion
                      draw->rcItem.right-draw->rcItem.left + 1,
```

File: dll/cpl/desk/settings.c [hunk 3]
```diff
@@ -421,6 +426,40 @@ SettingsOnInitDialog(IN HWND hwndDlg)
...
+                      draw->rcItem.bottom-draw->rcItem.top+1,
```
Comment (katahiromz) [inline dll/cpl/desk/settings.c:453]:
```suggestion
                      draw->rcItem.bottom-draw->rcItem.top + 1,
```

File: dll/cpl/desk/desk.c (hunk 1 above)
Comment (binarymaster) [reply dll/cpl/desk/desk.c]: Fixed.

File: dll/cpl/desk/desk.c [hunk 4]
```diff
@@ -197,6 +198,16 @@ DisplayApplet(HWND hwnd, UINT uMsg, LPARAM wParam, LPARAM lParam)
     g_GlobalData.pwszAction = pwszAction;
     g_GlobalData.desktop_color = GetSysColor(COLOR_DESKTOP);
 
+    /* Initialize the monitor preview bitmap, used on multiple pages */
+    g_GlobalData.hMonitorBitmap = LoadBitmapW(hApplet, MAKEINTRESOURCEW(IDC_MONITOR));
+    if (g_GlobalData.hMonitorBitmap != NULL)
+    {
+        GetObject(g_GlobalData.hMonitorBitmap, sizeof(bitmap), &bitmap);
```
Comment (HBelusca) [inline dll/cpl/desk/desk.c:205]:
```suggestion
        GetObjectW(g_GlobalData.hMonitorBitmap, sizeof(bitmap), &bitmap);
```

File: dll/cpl/desk/desk.h [hunk 5]
```diff
@@ -63,6 +63,20 @@ INT AllocAndLoadString(LPTSTR *lpTarget,
 
 ULONG __cdecl DbgPrint(PCCH Format,...);
 
+/* The values in these macros are dependent on the
```
Comment (HBelusca) [inline dll/cpl/desk/desk.h:66]:
```suggestion
/*
 * The values in these macros are dependent on the
```

File: dll/cpl/desk/desk.h [hunk 6]
```diff
@@ -63,6 +63,20 @@ INT AllocAndLoadString(LPTSTR *lpTarget,
...
+ * layout of the monitor image and they must be adjusted
+ * if that image will be changed.
```
Comment (HBelusca) [inline dll/cpl/desk/desk.h:68]:
```suggestion
 * if that image is changed.
```

File: dll/cpl/desk/settings.c [hunk 7]
```diff
@@ -377,6 +380,8 @@ SettingsOnInitDialog(IN HWND hwndDlg)
 
         OnDisplayDeviceChanged(hwndDlg, pData, pData->DisplayDeviceList);
 
+        ShowWindow(GetDlgItem(hwndDlg, IDC_RESOLUTION_PREVIEW), SW_HIDE);
```
Comment (HBelusca) [inline dll/cpl/desk/settings.c:383]: Are the controls correctly shown again *if* one adds or removes at runtime a monitor while keeping the desk.cpl tab opened? (I don't know whether Windows updates that at runtime as well :thinking:)

Comment (binarymaster) [reply dll/cpl/desk/settings.c:383]: I have checked Windows desk.cpl, it does not handle resolution changes in runtime (i.e. when the corresponding page already has initialzied), it probably does not handle monitor hotplugging as well (I don't know any emulator supporting that feature).

File: dll/cpl/desk/desk.c (hunk 4 above)
Comment (binarymaster) [reply dll/cpl/desk/desk.c]: Addressed.

File: dll/cpl/desk/settings.c (hunk 7 above)
Comment (HBelusca) [reply dll/cpl/desk/settings.c:383]: Note:
1. on windows, the equivalent functionality touched in this PR is in themeui.dll (loaded and displayed by desk.cpl afterwards)
2. windows doesn't update the page even if you go to another page, and then go back?

Comment (binarymaster) [reply dll/cpl/desk/settings.c:383]: 1. Yes, this is correct. This is also why you need both desk.cpl/themeui.dll from XP when testing in ReactOS.
2. Yes, it doesn't react to the changes even when switching pages.

**Review by katahiromz — APPROVED** [review]

**Pull request merged**